services/src/main/java/org/keycloak/services/resources/admin/RealmsAdminResource.java

```diff
@@ -174,9 +174,8 @@ public Response importRealm(InputStream requestBody) {
 
             return Response.created(location).build();
         } catch (ModelDuplicateException e) {
-            logger.error("Conflict detected", e);
             if (session.getTransactionManager().isActive()) session.getTransactionManager().setRollbackOnly();
-            throw ErrorResponse.exists("Conflict detected. See logs for details");
+            throw ErrorResponse.exists("Realm with same name already exists");
```

Comment:
There is `KeycloakErrorHandler.getResponse()` that should already handle any `ModelException`. 

So I suggest to remove the code of catching `ModelDuplicateException`, `ModelIllegalStateException` and `ModelException` altogether. Looking at the code, I am a bit worried that sometimes the transaction is rolled back, and sometimes not. 

If you feel in the mood, it might be good to deprecate the `ModelDuplicateException` with a no-arg constructor so we enfore that we always set an error message that can be shown to the user.

Reply:
Thanks for the feedback, @ahus1 ! I've applied all your suggestions

Reply:
It seems that my previous comment was only partially correct: While the KeycloakErrorHandler recognizes at least some of the exceptions, it maps a plain `ModelException` to a server error and not to a client error.  

So it might be necessary to bring back 

```
} catch (ModelException e) {
       throw ErrorResponse.error(e.getMessage(), Response.Status.BAD_REQUEST);
}
```

to avoid regressions. Treating all `ModelException`s as bad requests is likely beyond this PR, and would be a breaking change in a minor release to consumers. Possibly something for KC 27

Reply:
Yeah, I came to the same conclusion. I was spending some time trying to find out a way to fix how the model exceptions are handled. But as you said, it is well beyond the scope of this PR.

Reply:
Not related to this PR, but indicates we seem to have a broader issue handling exceptions